packages/toolkit/src/query/react/exceptions.ts

```diff
@@ -0,0 +1,29 @@
+const computeErrorMessage = (reason: any, queryKey: string) => {
+  let message = `rtk-query suspense error ${queryKey}: `
+
+  if (reason instanceof Error) {
+    message += reason
+  } else if (typeof reason === 'object' && reason !== null) {
+    ;[reason?.status, reason?.code, reason?.error].forEach((value) => {
+      if (value) {
+        message += ` ${value}`
+      }
+    })
+  } else {
+    message += reason
+  }
+
+  return message
+}
+
+export class SuspenseQueryError extends Error {
+  constructor(
+    public reason: unknown,
+    public endpointName: string,
+    public retryQuery: () => void
+  ) {
+    super(computeErrorMessage(reason, endpointName))
+    this.reason = reason
+    this.name = 'SuspenseQueryError'
```

Comment:
```suggestion
    super(computeErrorMessage(reason, endpointName))
    this.reason = reason
    this.name = 'SuspenseQueryError'
    Object.setPrototypeOf(this, SuspenseQueryError.prototype);
```

see https://www.typescriptlang.org/docs/handbook/2/classes.html#inheriting-built-in-types

Reply:
https://github.com/reduxjs/redux-toolkit/pull/2245/files#diff-8b83915d0409b42b46a5dfdc3c600a5210168bd1ca5165fa0598ae4f820a66acR31